### PR TITLE
Also listened to 'add' event for chokidar file watchers

### LINKED ISSUE #161: --watch mode doesn't pick up new files
Running `postcss -u stylelint -u postcss-reporter --no-map --watch -d ./src ./src/**/*.css` in a directory structure with `src/index.css`:
* Changes to `src/index.css` are picked up fine
* Adding `src/new.css` and then modifying it doesn't get picked up

Is this by design? Can it be fixed?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Moves the 'change' callback to a separate onAddOrChange that now is also passed to 'add'.

Fixes #161.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
Err. CI is complaining about lines I didn't modify...

File: index.js [hunk 1]
```diff
@@ -90,28 +90,31 @@ Promise.resolve()
 
       if (config.file) watcher.add(config.file)
 
-      watcher
-        .on('ready', () => {
-          console.warn(chalk.bold.cyan('Waiting for file changes...'))
-        })
-        .on('change', file => {
-          let recompile = []
+      const onAddOrChange = file => {
```
Comment (michael-ciniawsky) [inline index.js]: ` const onAddOrChange` => `const recompile`

File: index.js [hunk 2]
```diff
@@ -90,28 +90,31 @@ Promise.resolve()
...
+        let recompile = []
```
Comment (michael-ciniawsky) [inline index.js]: `let recompiled` => `let changed`

**Comment by michael-ciniawsky** [comment]
Travis is definitely unrelated (something weird with `prettier`), this should be rumberstamp good to go

@RyanZim Does this require changes for `depGraph` aswell ?

**Comment by RyanZim** [comment]
`prettier` errors are not unrelated; the linter is erroring out because the indentation is wrong. Run `npm test` locally, and prettier will auto-format your code. Commit the changes and squash all 3 commits into one commit, and force-push.

This also should have tests.

**Comment by RyanZim** [comment]
Ignore my previous deleted comment. Will fix prettier errors ASAP.

**Comment by RyanZim** [comment]
Fixed, please rebase on `master`.

**Comment by JoshuaKGoldberg** [comment]
To be honest, I'm no longer using this tool, and development doesn't work on my OS (#164). I'm not going to have the time to investigate the single CI test failure. Hopefully this PR ends up being of some use to you!

**Comment by michael-ciniawsky** [comment]
Could you nevertheless rebase, so this can be merged ?  Does 'only' the current tests fail on windows or doesn't it work at all ?

**Comment by JoshuaKGoldberg** [comment]
On Windows, none of the tests work when I run `npm run ci` on master. On CI for this pull request, there's an introduced failure.

Out of curiosity, why the requirement to rebase? Do you not want PRs to squash commits?

**Comment by michael-ciniawsky** [comment]
The watch test are definitely flaky

**Comment by RyanZim** [comment]
> The watch test are definitely flaky

Yeah, they drive me nuts. Gotta figure out something sometime.

**Comment by RyanZim** [comment]
> Out of curiosity, why the requirement to rebase? Do you not want PRs to squash commits?

@JoshuaKGoldberg I always squash commits; I just wanted to see the CI rerun with the prettier issue fixed.

NVM the failing watch tests; they are unpredictably flacky.

**Comment by RyanZim** [comment]
Finally got the CI to pass after restarting it for the nth time. :grimacing: Gotta do some quick tests locally before I merge.

**Comment by RyanZim** [comment]
This PR doesn't work for me; since we're passing the de-globbed file list into chokidar. It's possible to fix this, but I've got to move carefully to ensure I don't break my smart recompile logic.

**Comment by JoshuaKGoldberg** [comment]
I've stopped using the tool and don't really have time for this PR. Best of luck! :)

**Pull request closed without merging**